peerdiscovery.go

```diff
@@ -256,15 +256,17 @@ func (p *peerDiscovery) listen() (recievedBytes []byte, err error) {
 
 		// log.Println(src, hex.Dump(buffer[:n]))
 
+		ip := strings.Split(src.String(), ":")[0]
 		p.Lock()
-		if _, ok := p.received[strings.Split(src.String(), ":")[0]]; !ok {
-			p.received[strings.Split(src.String(), ":")[0]] = buffer[:n]
+		if _, ok := p.received[ip]; !ok {
+			p.received[ip] = buffer[:n]
 		}
+		p.Unlock()
+		p.RLock()
 		if len(p.received) >= p.settings.Limit && p.settings.Limit > 0 {
-			p.Unlock()
+			p.RUnlock()
 			break
 		}
-		p.Unlock()
```

Comment:
This can't be deleted - it should be changed to p.RUnlock, otherwise the locks are stuck

Reply:
you are right. It's my mistaken.